### PR TITLE
Feature/resolver property name fallback

### PR DESCRIPTION
**Pull request by Hendy**
PR for issue #74

### CODE REVIEW

File: src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs [hunk 1]
```diff
@@ -28,12 +28,21 @@ public class AppSettingValueResolver : DittoValueResolver<AppSettingAttribute>
         /// </returns>
         public override object ResolveValue(ITypeDescriptorContext context, AppSettingAttribute attribute, CultureInfo culture)
         {
-            if (string.IsNullOrWhiteSpace(attribute.AppSettingKey))
+            // default to attribute value
+            string appSettingKey = attribute.AppSettingKey;
```
Comment (leekelleher) [inline src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs]: I think `PropertyDescriptor` is never null, (but I could be wrong? @mattbrailsford?).

On another note, we could get this into a shorter statement, as we do in [`UmbracoPropertyValueResolver.cs` line 33](https://github.com/leekelleher/umbraco-ditto/blob/develop/src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/UmbracoPropertyValueResolver.cs#L33)

    var umbracoPropertyName = attribute.PropertyName ?? (context.PropertyDescriptor != null ? context.PropertyDescriptor.Name : string.Empty);

*(not that we're aiming for crazy single-lines-of-code)*

Comment (mattbrailsford) [inline src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs]: I can't think of any reason PropertyName would ever be null, so you could probably just use

`var key = attribute.PropertyName ?? context.PropertyDescriptor;`

Comment (Hendy) [inline src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs]: @leekelleher I thought the same about context.PropertyDescriptor never being null, but took the lead from the line you mention in UmbracoPropertyValueResolver

Comment (mattbrailsford) [inline src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs]: in all honesty, it doesn't hurt to be over defensive :)

Comment (leekelleher) [inline src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs]: Thanks @mattbrailsford @Hendy ... sure, null-reference checking is fine, (I'm usually an over-defensive coder :bowtie:)

@Hendy Did you want to amend the code? I'm happy to merge in whenever (we can pick up small tidy-ups later)

Comment (Hendy) [inline src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs]: happy to update it - is it safe to assume context.PropertyDescriptor.Name will never be null ?

Comment (leekelleher) [inline src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs]: Argh! Let's assume that it *might* be `null` :wink:  let's keep the check in.

**Comment by leekelleher** [comment]
Hey @Hendy, I'm going to merge this in now... I'm aiming to get the 0.7.0 release out tomorrow, so gonna do some tidy up this evening.

**Pull request merged**